test/helpers/mocked_cache_backend.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Allows to easily simulate a single broken call to backend
+# with a passthrough for everything else.
+class MockedCacheBackend < SimpleDelegator
+  def write(key, value, options = {})
+    if stubbed_calls[0] == [key, value]
+      stubbed_calls.shift
+      return false
+    end
+
+    super
+  end
+
+  def stub_call(key, value)
+    stubbed_calls << [key, value]
+  end
+
+  def stubbed_calls
+    @stubbed_calls ||= []
+  end
+end
```

Comment:
@dylanahsmith usually with rspec I would go with something like 
```ruby
expect(backend).to receive(:write).and_return(false).once
expect(backend).to receive(:write).with(...).and_call_original
```

but from what I've checked mocha doesn't have anything like this, so I've decided to go with simple delegator instead.